tests/Owin.Security.OpenIdConnect.Server.Tests/OAuth2AuthorizationClientCredentialsGrantTests.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Owin.Testing;
+using Xunit;
+
+namespace Owin.Security.OpenIdConnect.Server.Tests {
+    public class OAuth2AuthorizationClientCredentialsGrantTests {
+
+        [Fact]
+        public async Task ClientCredentials_MissingCredentials_BadRequest() {
+            var server = TestServer.Create(app => {
+                app.Properties["host.AppName"] = "Owin.Security.OpenIdConnect.Server.Tests";
+                app.UseOpenIdConnectServer(new OpenIdConnectServerOptions {
+                    SystemClock = new TestClock()
+                });
+            });
+
+            var response = await server.HttpClient.PostAsync(
+                "https://example.com/connect/token",
+                new StringContent("scope=openid&grant_type=client_credentials", Encoding.UTF8, "application/x-www-form-urlencoded"));
+
+            Assert.Equal(HttpStatusCode.BadRequest, response.StatusCode);
+            Assert.Equal("invalid_client", (await response.GetJsonBodyAsync()).Value<string>("error"));
```

Comment:
Using the `OpenIdConnectMessage` primitive for both the request and the response might be easier.